### PR TITLE
Source BigQuery: Fix NPE when `dataset_id` isn't provided

### LINKED ISSUE #5326: BigQuery source: "Could not connect with provided configuration. Error: null"
## Enviroment
- **Airbyte version**:  0.29.4-alpha
- **OS Version / Instance**: AWS EC2
- **Deployment**: Docker
- **Source Connector and version**: source-bigquery:0.1.1 
- **Severity**: Critical
- **Step where error happened**: Setup new connection

## Current Behavior
When setting up BigQuery as a new source I get the following error:
```
Could not connect with provided configuration. Error: null
```
The same configuration worked just fine when used to set up BigQuery as a destination

## Expected Behavior
Source connected successfully

## Logs
```
2021-08-11 04:27:50 INFO () TemporalAttemptExecution(get):110 - Executing worker wrapper. Airbyte version: 0.29.4-alpha
2021-08-11 04:27:50 INFO () LineGobbler(voidCall):85 - Checking if airbyte/source-bigquery:0.1.1 exists...
2021-08-11 04:27:50 INFO () LineGobbler(voidCall):85 - airbyte/source-bigquery:0.1.1 was found locally.
2021-08-11 04:27:50 INFO () DockerProcessFactory(create):146 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/4bf35597-e260-4815-8173-21ef35323e20/0 --network host --log-driver none airbyte/source-bigquery:0.1.1 check --config source_config.json
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 INFO i.a.i.s.b.BigQuerySource(main):183 - {} - starting source: class io.airbyte.integrations.source.bigquery.BigQuerySource
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 INFO i.a.i.b.IntegrationRunner(run):96 - {} - Running integration: io.airbyte.integrations.source.bigquery.BigQuerySource
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 INFO i.a.i.b.IntegrationCliParser(parseOptions):135 - {} - integration args: {check=null, config=source_config.json}
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 INFO i.a.i.b.IntegrationRunner(run):100 - {} - Command: CHECK
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 INFO i.a.i.b.IntegrationRunner(run):101 - {} - Integration config: IntegrationConfig{command=CHECK, configPath='source_config.json', catalogPath='null', statePath='null'}
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 WARN c.n.s.JsonMetaSchema(newValidator):338 - {} - Unknown keyword airbyte_secret - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 2021-08-11 04:27:52 INFO i.a.i.s.r.AbstractRelationalDbSource(check):154 - {} - Exception while checking connection: 
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - java.lang.NullPointerException: null
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 	at io.airbyte.integrations.source.bigquery.BigQuerySource.toDatabaseConfig(BigQuerySource.java:73) ~[io.airbyte.airbyte-integrations.connectors-source-bigquery-0.28.1-alpha.jar:?]
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 	at io.airbyte.integrations.source.relationaldb.AbstractRelationalDbSource.createDatabaseInternal(AbstractRelationalDbSource.java:490) ~[io.airbyte.airbyte-integrations.connectors-source-relational-db-0.28.1-alpha.jar:?]
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 	at io.airbyte.integrations.source.relationaldb.AbstractRelationalDbSource.check(AbstractRelationalDbSource.java:147) [io.airbyte.airbyte-integrations.connectors-source-relational-db-0.28.1-alpha.jar:?]
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 	at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:122) [io.airbyte.airbyte-integrations.bases-base-java-0.28.1-alpha.jar:?]
2021-08-11 04:27:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):73 - 	at io.airbyte.integrations.source.bigquery.BigQuerySource.main(BigQuerySource.java:184) [io.airbyte.airbyte-integrations.connectors-source-bigquery-0.28.1-alpha.jar:?]
2021-08-11 04:27:52 INFO () TemporalAttemptExecution(get):133 - Stopping cancellation check scheduling...
```

### PR DESCRIPTION
**Pull request by 69mb**
## What
Fixes #5326

## How
`dataset_id` is an optional config parameter so made change to store 
and handle it as empty string.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-bigquery/src/test/resources/test_config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "dataset_id": "dataset",
+  "project_id": "project",
+  "credentials_json": "credentials"
+}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bigquery/src/test/resources/test_config.json:5]: EOF, can you apply `./gradlew format` to ensure everything is formatted corretly?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
thanks @69mb small comment on formatting files. I'll review later today!

File: airbyte-integrations/connectors/source-bigquery/src/test/resources/test_config.json (hunk 1 above)
Comment (69mb) [reply airbyte-integrations/connectors/source-bigquery/src/test/resources/test_config.json]: Ok, done.

**Comment by marcosmarxm** [comment]
Hi @69mb can you solve the conflict and apply `./gradlew format`? I'll run tests and check your code later today.

**Review by marcosmarxm — APPROVED** [review]
LGTM! Thanks @69mb

**Pull request merged**